src/home_robot/home_robot/agent/ovmm_agent/ovmm_agent.py

```diff
@@ -228,84 +192,141 @@ def _hardcoded_place(self):
+            )
+        new_state = None
+        if action == DiscreteNavigationAction.STOP:
+            action = None
+            new_state = Skill.GAZE_AT_OBJ
+        return action, info, new_state
+
+    def _gaze_at_obj(
+        self, obs: Observations, info: Dict[str, Any]
+    ) -> Tuple[DiscreteNavigationAction, Any, Optional[Skill]]:
+        if self.skip_skills.gaze_at_obj:
+            term = True
+        else:
+            action, term = self.gaze_agent.act(obs)
+        new_state = None
+        if term:
+            # action = (
+            #     {}
+            # )  # TODO: update after simultaneous gripping/motion is supported
+            # action["grip_action"] = [1]  # grasp the object when gaze is done
+            # self.is_pick_done[0] = 1
+            action = None
+            new_state = Skill.PICK
+        return action, info, new_state
+
+    def _pick(
+        self, obs: Observations, info: Dict[str, Any]
+    ) -> Tuple[DiscreteNavigationAction, Any, Optional[Skill]]:
+        if self.skip_skills.pick:
+            action = None
+        elif self.config.AGENT.SKILLS.PICK.type == "oracle":
+            pick_step = self.timesteps[0] - self.pick_start_step[0]
+            if pick_step == 0:
+                action = DiscreteNavigationAction.SNAP_OBJECT
+            elif pick_step == 1:
+                action = None
+            else:
+                raise ValueError(
+                    "Still in oracle pick. Should've transitioned to next skill."
+                )
+        new_state = None
+        if action is None:
+            new_state = Skill.NAV_TO_REC
+        return action, info, new_state
+
+    def _nav_to_rec(
+        self, obs: Observations, info: Dict[str, Any]
+    ) -> Tuple[DiscreteNavigationAction, Any, Optional[Skill]]:
+        nav_to_rec_type = self.config.AGENT.SKILLS.NAV_TO_REC.type
+        if self.skip_skills.nav_to_rec:
+            action = DiscreteNavigationAction.STOP
+        elif nav_to_rec_type == "heuristic":
+            action, info = self._heuristic_nav(obs, info)
+        elif nav_to_rec_type == "rl":
+            raise NotImplementedError
+        else:
+            raise ValueError(
+                f"Got unexpected value for NAV_TO_REC.type: {nav_to_rec_type}"
+            )
+        new_state = None
+        if action == DiscreteNavigationAction.STOP:
+            action = None
+            new_state = Skill.GAZE_AT_REC
+        return action, info, new_state
+
+    def _gaze_at_rec(
+        self, obs: Observations, info: Dict[str, Any]
+    ) -> Tuple[DiscreteNavigationAction, Any, Optional[Skill]]:
+        if self.skip_skills.gaze_at_rec:
+            term = True
+        else:
+            action, term = self.gaze_agent.act(obs)
+        new_state = None
+        if term:
+            action = None
+            new_state = Skill.PLACE
+        return action, info, new_state
+
+    def _place(
```

Comment:
We might want to add a short docstring for each of these

Reply:
I think each of the skill methods are small and self-explanatory. I added a comment above all the skill methods to explain the general structure of all of the methods.